### PR TITLE
130 remove global theme settings

### PR DESCRIPTION
**Pull request by jenlampton**
PR for https://github.com/backdrop/backdrop-issues/issues/130

### CODE REVIEW

**Comment by jenlampton** [comment]
These tests fail on travis, due to https://github.com/backdrop/backdrop/pull/460.  As soon as that's fixed we should be good to go here.

**Comment by quicksketch** [comment]
Could you prefix these commits with the issue number?

File: core/modules/system/system.admin.inc [hunk 1]
```diff
@@ -1497,11 +1377,76 @@ function system_site_information_settings_validate($form, &$form_state) {
...
+  if (empty($form_state['values']['site_default_favicon']) && !empty($form_state['values']['site_favicon_path'])) {
+    $form_state['values']['favicon_mimetype'] = file_get_mimetype($form_state['values']['site_favicon_path']);
+  }
+
   config('system.site')
     ->set('site_name', $form_state['values']['site_name'])
+    ->set('site_default_logo', $form_state['values']['site_default_logo'])
+    ->set('site_logo_path', $form_state['values']['site_logo_path'])
+    ->set('site_default_favicon', $form_state['values']['site_default_favicon'])
+    ->set('site_favicon_path', $form_state['values']['site_favicon_path'])
```
Comment (quicksketch) [inline core/modules/system/system.admin.inc:1449]: site_favicon_mimetype needs to be saved as configuration here.

File: core/includes/theme.inc [hunk 2]
```diff
@@ -1437,25 +1437,13 @@ function theme_get_setting($setting_name, $theme = NULL) {
...
-    // To add new global settings, add their default values below, and then
-    // add form elements to system_theme_settings() in system.admin.inc.
     $cache[$theme] = array(
-      'default_logo'                     =>  1,
-      'logo_path'                        =>  '',
-      'default_favicon'                  =>  1,
-      'favicon_path'                     =>  '',
       // Use the IANA-registered MIME type for ICO files as default.
       'favicon_mimetype'                 =>  'image/vnd.microsoft.icon',
     );
```
Comment (quicksketch) [inline core/includes/theme.inc]: This mimetype line should be removed and added to the system.site.json config file as the default.

File: core/modules/system/system.admin.inc [hunk 3]
```diff
@@ -1497,11 +1377,76 @@ function system_site_information_settings_validate($form, &$form_state) {
...
+  if (!empty($form_state['values']['site_logo_path'])) {
+    $form_state['values']['site_logo_path'] = _system_site_information_settings_validate_path($form_state['values']['site_logo_path']);
+  }
+  if (!empty($form_state['values']['site_favicon_path'])) {
+    $form_state['values']['site_favicon_path'] = _system_site_information_settings_validate_path($form_state['values']['site_favicon_path']);
+  }
+
+  if (empty($form_state['values']['site_default_favicon']) && !empty($form_state['values']['site_favicon_path'])) {
+    $form_state['values']['favicon_mimetype'] = file_get_mimetype($form_state['values']['site_favicon_path']);
+  }
```
Comment (quicksketch) [inline core/modules/system/system.admin.inc:1442]: For consistency, this should probably be:

```
$form_state['values']['site_favicon_mimetype'] = file_get_mimetype($form_state['values']['site_favicon_path']);
```

And then save this from form_state below.

File: core/modules/system/system.install [hunk 4]
```diff
@@ -1792,6 +1792,22 @@ function system_update_1013() {
...
+  if (array_key_exists($theme_settings['logo_path'])) {
+    $config->set('site_logo_path', $theme_settings['logo_path']);
+  }
+  if (array_key_exists($theme_settings['default_favicon'])) {
+    $config->set('site_default_favicon', $theme_settings['default_favicon']);
+  }
+  if (array_key_exists($theme_settings['favicon_path'])) {
+    $config->set('site_favicon_path', $theme_settings['favicon_path']);
+  }
+
```
Comment (quicksketch) [inline core/modules/system/system.install:1813]: Needs site_favicon_mimetype. All of these settings should also be added system.site.json in the default provided files so they exist on new site installs.

**Comment by quicksketch** [comment]
Looks like this is being split up into multiple PRs. The first replacement it at https://github.com/backdrop/backdrop/pull/464.

**Comment by jenlampton** [comment]
Closing this PR in favor of https://github.com/backdrop/backdrop/pull/464A

**Comment by jenlampton** [comment]
Not sure why this was reopened, the other one should suffice.

**Pull request closed without merging**